ion-schema/src/test/kotlin/com/amazon/ionschema/model/SchemaDocumentTest.kt

```diff
@@ -0,0 +1,147 @@
+package com.amazon.ionschema.model
+
+import com.amazon.ion.system.IonSystemBuilder
+import com.amazon.ionschema.IonSchemaVersion
+import com.amazon.ionschema.model.SchemaDocument.Item
+import org.junit.jupiter.api.Assertions.assertEquals
+import org.junit.jupiter.api.Assertions.assertIterableEquals
+import org.junit.jupiter.api.Assertions.assertNull
+import org.junit.jupiter.api.Assertions.assertSame
+import org.junit.jupiter.api.Assertions.assertTrue
+import org.junit.jupiter.api.Test
+
+@ExperimentalIonSchemaModel
+class SchemaDocumentTest {
+
+    val ION = IonSystemBuilder.standard().build()
+
+    @Test
+    fun `test get header when no header present`() {
+        val schema = SchemaDocument("schema.isl", listOf())
+        assertNull(schema.header)
+    }
+
+    @Test
+    fun `test get header when header present`() {
+        val header = Item.Header()
+        val footer = Item.Footer()
+        val schema = SchemaDocument("schema.isl", listOf(header, footer))
+        assertSame(header, schema.header)
+    }
+
+    @Test
+    fun `test get footer when no footer present`() {
+        val schema = SchemaDocument("schema.isl", listOf())
+        assertNull(schema.footer)
+    }
+
+    @Test
+    fun `test get footer when footer present`() {
+        val header = Item.Header()
+        val footer = Item.Footer()
+        val schema = SchemaDocument("schema.isl", listOf(header, footer))
+        assertSame(footer, schema.footer)
+    }
+
+    @Test
+    fun `test get declaredTypes when no types present`() {
+        val schema = SchemaDocument("schema.isl", listOf())
+        assertTrue(schema.declaredTypes.isEmpty())
+    }
+
+    @Test
+    fun `test get declaredTypes when some types present`() {
+        val type0 = NamedTypeDefinition(
+            "type0",
+            TypeDefinition(
+                constraints = listOf(),
```

Comment:
(Suggestion) Can we please add some tests for types that have constraints defined for ISL 1.0 and ISL 2.0.?

Reply:
My plan is to have builders that only allow you to create valid 1.0 or 2.0 types, so it will not be possible to create something with incompatible constraints. When I have those, it will make sense to test them in the way you are suggesting.

Reply:
I couldn't find any type tests with constraints in them. I was suggesting to have some tests for types with ISL 1.0 constraints and other tests for types with  ISL 2.0 constraints.

Reply:
I see. There are no tests for types in the data model because `TypeDefinition` doesn't really have functionality to test. It's just plain data.

Reply:
Oh, but you are still asserting two type definitions below in this test. I think there should be some tests for checking how the data classes defined in `Constraint` work whether its separate tests for constraints  or as part of tests for types with constraints. If you have planned for this in a follow up PR then I am okay with what we have now but it might be worth mentioning this in an issue/TODO.

Reply:
> Oh, but you are still asserting two type definitions below in this test. I think there should be some tests for checking how the data classes defined in Constraint work whether its separate tests for constraints or as part of tests for types with constraints.

Okay, so any tests regarding constraints would be done separately from this. This test class is unit tests for `SchemaDocument`, so the scope is limited to things that interact directly with `SchemaDocument`.

I am creating some empty types so that I have something to put in the schema, and I am asserting that they are the _same object_ (referential equality rather than value-based equality). Instead of using empty/dummy types, I can use mocks if that makes it more clear.

> If you have planned for this in a follow up PR then I am okay with what we have now but it might be worth mentioning this in an issue/TODO.

When there is functionality to be tested, I will either write the tests or I will create an issue/todo. Right now, there is nothing to be unit tested for `TypeDefinition`, so I don't even know what I would put into the issue/todo.

Reply:
Thanks for explaining. It seems like these tests are just to ensure `SchemaDocument` is constructed properly.